proposals/0458-strict-memory-safety.md

```diff
@@ -122,10 +121,10 @@ The operation `UnsafeMutableBufferPointer.swapAt` swaps the values at the given
 
 ```swift
 extension UnsafeMutableBufferPointer {
-  @unsafe public func swapAt(_ i: Element, _ j: Element) {
+  @unsafe public func swapAt(_ i: Index, _ j: Index) {
     guard i != j else { return }
     precondition(i >= 0 && j >= 0)
-    precondition(unsafe i < endIndex && j < endIndex)
+    precondition(i < endIndex && j < endIndex)
```

Comment:
According to the current proposal, the `unsafe` is needed with the `endIndex` because of the implicit `self` being of an unsafe type. There is discussion on the review thread about how we might avoid that.